### DISCUSSION TITLE
Wrong scheule settings for AVATTO ME167

### DISCUSSION THREAD

**Discussion by rolf-achim** [Koenkk/zigbee2mqtt#19639] [Devices]
Hi. I'm located in Germany and new to ZigBeeMQTT. I'm running ZigBee2MQTT version 1.33.1 on a windows computer and successfully joined a TRV Avatto ME167. Everything is working like a charm. However, if I'm setting schedules for specific days, they are executed at a different day.
Example:
I configure the MQTT topic "zigbee2mqtt.0.0x84ba20fffe9a64fa.schedule_thursday" with "07:04/21.0 08:04/25.4 19:04/23.4 22:04/18.0",
the ZigBee response I can see in the debug log is
"Zigbee2MQTT:debug 2023-11-09 17:21:27: Received Zigbee message from 'Heizkörperthermostat_Keller', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[4,7,4,0,210,8,4,0,254,19,4,0,234,22,4,0,180],"type":"Buffer"},"datatype":0,"dp":31}],"seq":26880}' from endpoint 1 with groupID 0"
I assume the leading "4" is indicating the fourth day of the week which should be Thursday.
But the setting will be executed from the device on Saturday. I have observed the same two-day shift for every weekday.
Where in the code can I change the weekday mapping from MQTT to the correct position in the device' buffer?

**Comment by joba-1** [comment]
2 years late, but now I have the same issue with my new installation:

It is monday and right now my thermostat switched temperature like on saturday or sunday:

```
Received 08:46:59

    QoS: 0
    Payload:

    battery_low: null
    child_lock: UNLOCK
    current_heating_setpoint: 20.5
    error: null
    frost_protection: null
    last_seen: '2025-02-24T08:46:59+01:00'
    linkquality: 65
    local_temperature: 20
    local_temperature_calibration: 3
    running_state: heat
    scale_protection: 'ON'
    schedule_friday: 05:30/20.5 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0
    schedule_monday: 05:30/20.5 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0
    schedule_saturday: 07:30/20.5 09:00/18.0 09:00/18.0 09:00/18.0 09:00/18.0 09:00/18.0
    schedule_sunday: 07:30/20.5 09:00/18.0 09:00/18.0 09:00/18.0 09:00/18.0 09:00/18.0
    schedule_thursday: 05:30/20.5 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0
    schedule_tuesday: 05:30/20.5 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0
    schedule_wednesday: 05:30/20.5 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0
    system_mode: auto

Received 09:00:53

    QoS: 0
    Payload:

    battery_low: null
    child_lock: UNLOCK
    current_heating_setpoint: 18
    error: null
    frost_protection: null
    last_seen: '2025-02-24T09:00:53+01:00'
    linkquality: 69
    local_temperature: 20
    local_temperature_calibration: 3
    running_state: heat
    scale_protection: 'ON'
    schedule_friday: 05:30/20.5 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0
    schedule_monday: 05:30/20.5 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0
    schedule_saturday: 07:30/20.5 09:00/18.0 09:00/18.0 09:00/18.0 09:00/18.0 09:00/18.0
    schedule_sunday: 07:30/20.5 09:00/18.0 09:00/18.0 09:00/18.0 09:00/18.0 09:00/18.0
    schedule_thursday: 05:30/20.5 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0
    schedule_tuesday: 05:30/20.5 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0
    schedule_wednesday: 05:30/20.5 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0 07:00/18.0
    system_mode: auto
```

How can I fix this or diagnose this further?

**Reply by joba-1** [reply]
looks like a device bug. Different ME167 stay on different weekdays

**Reply by mqatgit** [reply]
I can confirm this. 3 devices, all have the weekday schedule order bug. And they have terrible zigbee connection stability, by the way, the only devices in my network that have any (and then: multiple per day) connectivity dropouts. But they always come back online, and mechanically, they work just fine.

**Reply by mqatgit** [reply]
By the way, the bug seems quite simple. The schedule items are sorted and *displayed* in alphabetical order in all interfaces, so the list is "Schedule friday", "Schedule monday", Schedule saturday" and so on, but internally, they apparently have a correct weekday sequence stored (starting with monday, I guess - you can find out if you give every weekday different 'auto' switching timings and then check logs what the devices actually do - I haven't fully done that).
That means that we also can work around this by translating displayed to stored weekdays as above... I'll try and see how that works.

**Reply by joba-1** [reply]
well, a user interface that sorts weekdays alphabetically itself has a bug :) 

I found a constant 2 day offset on mine i.E. monday schedule is used on wednesday and so on.

The proper fix would be to change the mapping in a custom herdsmen device - probsbly easy to do but had too low prio for me to look into it until now.

**Comment by phuddle1** [comment]
Same problem here. I ve seven ME167 with an offset for two days. The issue is still there.

**Comment by Offebaescher** [comment]
This is being discussed also here:
https://github.com/Koenkk/zigbee-herdsman-converters/pull/10561
And here:
https://github.com/koenkk/zigbee2mqtt/issues/29709

I confirm that i have this behaviour with schedule shifted by 2 days with Device Zigbee-Modell TS0601 _TZE200_hvaxb2tc (Thermostatic radiator valve), Modell TRV06_1 AVATTO

Running Zigbee2MQTT-Version 2.7.1, zigbee-herdsman-converters Version 25.83.1
